### PR TITLE
refactor(ios/engine): KMP installation now relies on KeymanPackage class, parses

### PR DESCRIPTION
**Pull request by jahorton**
This PR finally roots out some of that duplicated parsing code I referenced in previous PRs.  The KMP installation methods have now been retooled to use the `Codable`-based `KMPMetadata` parsing introduced by #3198 and its parent PRs, integrating it fully into the `KeymanPackage` class set.

While I want to further refactor the installation methods themselves, placing their core code in a more object-oriented location (`ResourceFileManager`), I figure that should be left for a later PR - there's one less set of moving parts this way.  I may want to implement some installation helpers on the `KeymanPackage` subclasses to simplify their code further at that time as well.

### User Testing Notes

As this actually _integrates_ the `Codable`-based parsing into the KeymanEngine's installation process, there's the slight chance that it may break installation for some .KMPs if they've been published while not following expected `kmp.json` formats.

Now, for which packages out there would be best to stress-test this with... I'm not sure.  But testing the limits of the kmp.json parsing by trying to install, uh, "unusual"/"unique" KMPs should be the best way to test this.

Personal test notes:  
- Installing "Salish, Straits" still works _mostly_ correctly.
  - Sometimes the followup lexical model install doesn't work right the first time, but that's been a persistent issue in need of followup.
    - It's likely to be affected by the upcoming cloud JS -> KMP rework, so I'm not stressed about it at this time.
  - If the chain-install doesn't work, the lexical model will install just fine from the language's "dictionary"-install menu.

### Unit Testing Notes

Currently passes all iOS unit tests.  (Manually run, since it's not continuously-integrated yet.)

This implies:
- `khmer_angkor.kmp` installs flawlessly
- `nrc.en.mtnt.model.kmp` installs flawlessly

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift [hunk 1]
```diff
@@ -613,191 +613,97 @@ public class Manager: NSObject, UIGestureRecognizerDelegate {
...
+          let oskPath = Storage.active.fontURL(forKeyboardID: keyboardID, filename: osk)
+          installableFiles.append([osk, oskPath])
+        }
+
+        if let font = k.font {
+          let displayPath = Storage.active.fontURL(forKeyboardID: keyboardID, filename: font)
+          installableFiles.append([font, displayPath])
+        }
+        do {
+          for item in installableFiles {
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:647]: Would it be better/possible to copy the entire folder?

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift [hunk 2]
```diff
@@ -613,191 +613,97 @@ public class Manager: NSObject, UIGestureRecognizerDelegate {
...
+    }
 
-        // Version uses a 'conditional initializer'.  If it fails, the version info is invalid.
-        if let _ = Version(version) {
-          // No problem
-        } else {
-          // Lazy-handle the error and replace version with 1.0.  Legacy decision from 2005.
-          version = "1.0"
-        }
+    for m in kmp.models {
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:675]: btw, there should only be one lexical model in a lexical model package

File: ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift [hunk 3]
```diff
@@ -40,8 +39,9 @@ public class KMPKeyboard: Codable {
     font = try values.decodeIfPresent(String.self, forKey: .font)
     isRTL = try values.decodeIfPresent(Bool.self, forKey: .isRTL) ?? false
     languages = try values.decode([KMPLanguage].self, forKey: .languages)
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift:41]: Does this lower-case the language ID's?
So we can match `str-Latn` == `str-latn`

File: ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift [hunk 4]
```diff
@@ -414,8 +414,6 @@ class ResourceDownloadQueue: HTTPDownloadDelegate {
         // The request has succeeded.
         if downloader!.requestsCount == 0 { // Download queue finished.
           let keyboards = batch.resources as! [InstallableKeyboard]
-
-          FontManager.shared.registerCustomFonts()
           log.info("Downloaded keyboard: \(keyboards[0].id).")
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift:417]: Is this for downloading cloud keyboards (.js) files and fonts from s.keyman.com?
Where did registerCustomFonts() get refactored to?

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift (hunk 1 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:647]: In another PR, yes.  This one is concerned with maintaining existing behavior while relying on a different parse source.

File: ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift (hunk 4 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift:417]: It's always called on `Manager.shared.addKeyboard()` calls, among other places.  This single call occurred before any new fonts actually got installed and was out of place.  I didn't touch any of its other calls.

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift (hunk 2 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:675]: While true for lexical models, it's not true for keyboards... and the next PR merges the control flow for both.  This sets up for that abstraction by highlighting how identical the two `parse___KMP` methods really are, even if one is technically more restricted than the other.

File: ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift (hunk 3 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift:41]: It does not do so yet, no.  :+1:

**Comment by darcywong00** [comment]
I'll approve this after the iOS test build passes
Current fail
> parse error: Invalid numeric literal at line 1, column 7
> [01:51:15]Process exited with code 4

File: ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift (hunk 3 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift:41]: Latest commit fixes this.

**Pull request merged**